raft/src/main/java/org/apache/kafka/snapshot/FileRawSnapshotReader.java

```diff
@@ -54,8 +54,12 @@ public Records records() {
     }
 
     @Override
-    public void close() throws IOException {
-        fileRecords.close();
+    public void close() {
```

Comment:
AutoCloseable#close throws Exception, so we can still throw IOException. Is there a reason why we want an unchecked exception here?

Reply:
Yeah. For consistency in the API. For example, most of the methods in this type perform an IO operation which can throw an `IOException` yet they have already been implemented to wrap them in a uncheck exception. In some places we wrap this on a unchecked exception is some places we don't. For the raft module, we have made an implicit decision to prefer uncheck exceptions for IO errors.

I tried to clean this up in https://github.com/apache/kafka/pull/10085:
> 5. Removed throws IOException from some methods. Some of types were inconsistently throwing IOException in some cases and throwing RuntimeException(..., new IOException(...)) in others. This PR improves the consistent by wrapping IOException in RuntimeException in a few more places and replacing Closeable with AutoCloseable.

This PR expands on that work/decision.

Reply:
Ok, sounds good. If we're strictly dealing with IOExceptions, maybe we can use UncheckedIOException?

Reply:
I created https://issues.apache.org/jira/browse/KAFKA-12773